src/services/utils.ts

```diff
@@ -178,45 +178,29 @@ export function replaceAll(string: string, replaceWhat: string, replaceWith: str
 }
 
 export function formatDownloadTitle(fileName: string, type: string | null, mime: string) {
-    if (!fileName) {
-        fileName = "untitled";
-    }
-
-    fileName = sanitize(fileName);
-
-    if (type === 'text') {
-        return `${fileName}.html`;
-    } else if (type && ['relationMap', 'canvas', 'search'].includes(type)) {
-        return `${fileName}.json`;
-    } else {
-        if (!mime) {
-            return fileName;
-        }
+  const fileNameBase = (!fileName) ? "untitled" : sanitize(fileName);
 
-        mime = mime.toLowerCase();
-        const filenameLc = fileName.toLowerCase();
-        const extensions = mimeTypes.extensions[mime];
+  const extension = (() => {
```

Comment:
I'm personally not very fond of IIFEs. I would suggest using a normal function inside this one or separate it out entirely and call it here.

Reply:
changes done
let me know, if that works for you this way (I did cheat a little bit to be honest), or if I should move it out fully :-)